ONSdigital/rm-tools
# Social test setup

# Motivation and Context

For social test 1 we need to run a social collection exercise without any UI. This PR is a set of scripts to run through the creation and setup of a social survey and collection exercise purely by API calls.

# What has changed



* Added scripts that hit the service endpoints directly to load and link data
* Added scripts to generate config for and run the collex-loader and event-loader tools to set up a social collection exercise and its events
* Added pyinvoke tasks to run all setup steps in sequence
* Added README documenting setup steps

# How to test?



To test locally:
* Start up all the services using [ras-rm-docker-dev](https://github.com/ONSdigital/ras-rm-docker-dev)
* Copy the example config for running locally from the README into your `.env` file in the `social-test-setup` directory
* Set `SAMPLE_PATH` to the path to your social sample file in the `.env` (see attachments on trello card for link to sample file)
* Run `make setup-and-execute` to run all the set up steps in sequence and execute the collection exercise it creates
* Tails the docker logs for the case service, action service or action exporter service, you should see the cases and actions being created without errors
* Check in `~/Documents/sftp`, social notification and pre-notification files should be generated (this may take a few minutes after execution and the files may be split into multiple parts depending on how often the action exporter scheduler is running)
* Check the notification and pre-notification files contain the addresses from your sample file, and the notification contains the IAC codes

If you want to watch the progress of individual setup steps, follow the README instructions for running the setup one step at a time.

# Links



https://trello.com/c/jOxn9kFI/104-dev-task-2-int-set-up-lms-survey-data-8
